src/FieldToken/index.jsx

```diff
@@ -31,7 +35,8 @@ const FieldToken = ({ label, onDismiss = noop, testId }) => {
         role="button"
         aria-label={`Remove ${label}`}
         tabIndex={0}
-        onClick={() => {
+        onClick={(e) => {
+          e.stopPropagation();
```

Comment:
We don't want this click to bubble up to any parent, as it may trigger another event. The dismissal click should be isolated to `FieldToken`